### DISCUSSION TITLE
german HL7 name.prefix.extension

### DISCUSSION THREAD

**Discussion by MichaelStrzata** [FirelyTeam/firely-net-sdk#2116] [Q&A] [answered]
Hallo, and sorry for  my english language...
My Name is Michael and i build Fhir with patient-data to work with Telematik-Infrastructur in germany. I like to build Fhir with the SDK, because ist fine and safe. Im new here and like to work with Fhir, im using the .NET SDK 4.0.0

Here i have some problems, and everyone i told my problem, i get a link to FIRE.LY to see, how it works with Birthdate/birthdateElement.
Unfortunaly ist works with Birthdate, with this example. Can someone show me, how will i add an extension to name.prefix?
There is the same element ( prefixElement ) but i cant add an extension.

![grafik](https://user-images.githubusercontent.com/107876190/174646060-af776dd7-702c-4874-9c93-9662522a0d4d.png)

But i need this...
![grafik](https://user-images.githubusercontent.com/107876190/174646304-10c06bdc-5cc5-4b14-8ba8-681a8fbc4450.png)

With FamilyElement it will add en Extension too, same like birthday...but not with HumanName.prefix.extension... 
and this problem i have similar with Address.line.extension. Here exists lineElement too, but it dont accept extension
![grafik](https://user-images.githubusercontent.com/107876190/174647131-029dd451-71ef-45ce-b19c-5b092182d5df.png)

Please dont link me to https://docs.fire.ly/projects/Firely-NET-SDK/model/extensions.html...   this isnt my problem...

i like to know, can SDK 4.0.0 humanName.prefix.Extension (and address.line.extension) and how can i get this...

thanx for the help

sinceraly Michael

**Comment by mbaltus** [comment, marked as answer]
Hello Michael,

You are running into your issue because the HumanName.prefix is a list of prefixes instead of a singular field.
This means that the AddExtension method is not present, since you need to add the extension to 1 item in the list. Here's a way to do that:

``` csharp
var hn = new HumanName();
var prefix = new FhirString("Prof.");
prefix.AddExtension("http://hl7.org/fhir/StructureDefinition/iso21090-EN-qualifier", new Code("AC"));
hn.PrefixElement.Add(prefix);

var prefix2 = new FhirString("Dr.");
prefix2.AddExtension("http://hl7.org/fhir/StructureDefinition/iso21090-EN-qualifier", new Code("AC"));
hn.PrefixElement.Add(prefix2);

yourPatient.Name.Add(hn);
```

You are supposed to create a list of prefixes if you have multiple ones. As you can see, I've only used "Prof." for this first prefix, and repeated the steps for each additional prefix so you get an actual list.

Hope this helps!

**Reply by MichaelStrzata** [reply]
Dear Mirjam
you save my day...so many questions and no help, and you tell me in 15 min a solution for my problem...you are an angel...
thanx and i wish you a beatiful day...
btw how time is it in your Country?  Your help ist really remarkable

Sinceraly Michael

**Reply by mbaltus** [reply]
You're welcome! I'm in CET, so it is evening here, but I saw your question come in and had some time. Happy coding!

**Reply by MichaelStrzata** [reply]
Hi Mirjam,
i got an another little issue... perhaps you can help me again?
In <meta><lastUpdated value="2021-04-06T08:30:00Z" /> i want to have this format...

i use -> LastUpdatedElement = Instant.FromLocalDateTime(Year(Now), Month(Now), Day(Now), Hour(Now), Minute(Now), Second(Now), 0)
to get this here -> <lastUpdated value="2022-06-23T11:48:46+02:00" /> and this looks great, but...
im not shure, if this correctly enought for german certification-buero. Can i write the value other?

Thanx and a beautiful day
Sinceraly Michael

**Reply by mbaltus** [reply]
A resource's meta.lastUpdated field is filled (or overwritten) by the server that stores the data, and not needed for resources that you send in to be stored. I do not know if you are a client or server side system.

Maybe you included some examples of what the output looks like, but they are not visible. I would actually make the code a bit more simple, since `Instant` has a `Now()` method:
`yourPatient.Meta.LastUpdatedElement = Instant.Now();`

Since this results in data that is compliant with the FHIR specification, I would expect that it will pass the German certification as well. At least it will if they have not added any extra requirements on the lastUpdated date.

**Reply by MichaelStrzata** [reply]
Hi Mirjam
i think so too... I dont use` Instant.now()`, because it will show milliseconds.  and youre right, in lastupdate the time will be writen from the server. this is my mistake ;-) ...I thought time on <timestamp value="2021-04-06T08:30:00Z"/>...this are my writingtime. I will write the time in UTC-format ...thanx for your help and a nice day
Sinceraly Michael
